walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/util/HttpAgentUtils.kt

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved
+
+package com.microsoft.walletlibrary.did.sdk.util
+
+import com.microsoft.walletlibrary.did.sdk.CorrelationVectorService
+import com.microsoft.walletlibrary.util.http.URLFormEncoding
+import javax.inject.Inject
+import javax.inject.Named
+
+/**
+ * Internal Class to apply common headers
+ */
+internal class HttpAgentUtils @Inject constructor(@Named("userAgentInfo") private val userAgentInfo: String,
+                                                  @Named("walletLibraryVersionInfo") private val walletLibraryVersionInfo: String,
+                                                  private val correlationVectorService: CorrelationVectorService) {
```

Comment:
It doesn't use the CorrelationVectorService. correlation vector is currently created for every network call using interceptor s. Doesn't it have to done here?

Reply:
This should have been called in the defaultHeaders call